src/python/pants/backend/python/pip_requirement.py

```diff
@@ -21,8 +22,35 @@ class PipRequirement:
     """
 
     @classmethod
-    def parse(cls, line: str) -> PipRequirement:
-        return cls(pkg_resources.Requirement.parse(line))
+    def parse(cls, line: str, description_of_origin: str = "") -> PipRequirement:
+        try:
+            return cls(pkg_resources.Requirement.parse(line))
+        except Exception as e:
+            scheme, netloc, path, query, fragment = urllib.parse.urlsplit(line)
+            if scheme and fragment.startswith("egg="):
```

Comment:
You'll want to be able to deal with more going on in the fragment since Pip can use subdirectory:
+ https://github.com/pantsbuild/pex/blob/904eef37fa36d738cc6409045239a7d80800e456/tests/test_requirements.py#L248
+ https://github.com/pantsbuild/pex/blob/904eef37fa36d738cc6409045239a7d80800e456/tests/test_requirements.py#L275-L276

And subdirectory is supported by Pex:
https://github.com/pantsbuild/pex/blob/904eef37fa36d738cc6409045239a7d80800e456/tests/integration/cli/commands/test_vcs_lock.py#L157-L168

Reply:
Hmm, what do we do with those other fragments in the converted URL? I suppose we can leave them all in and assume they will be used if needed and ignored if not needed.  

Experiments show that this works (e.g., leaving the fragment #egg=Django in `Django @ git+https://github.com/django/django.git@stable/2.1.x#egg=Django` works, and also the subdirectory gets used correctly in `sdev_logging_utils @ git+https://github.com/SerialDev/sdev_py_utils.git@bd4d36a0#egg=sdev_logging_utils&subdirectory=sdev_logging_utils`)

Reply:
BTW, re the comment [here](https://github.com/pantsbuild/pex/blob/904eef37fa36d738cc6409045239a7d80800e456/tests/integration/cli/commands/test_vcs_lock.py#L157-L168) re "This is a trick you cannot do with a direct reference VCS URL". Does "this" refer to specifying a subdirectory? 

Because that does seem to work with a direct reference VCS URL, in practice anyway. E.g., `sdev_logging_utils @ git+https://github.com/SerialDev/sdev_py_utils.git@bd4d36a0#egg=sdev_logging_utils&subdirectory=sdev_logging_utils` works (And omitting the `subdirectory` does not work)

Reply:
Aha. Gotcha. It works because Pip is in the pipeline and the PEP-440 is agnostic about URL content. Feed that PEP-440 to Poetry and it will fail (I'd bet).